src/_test_/App.test.js

```diff
@@ -23,37 +24,36 @@ describe("<App /> component", () => {
   });
 });
 
-describe('<App /> integration', () => {
-    test('renders a list of events matching the city selected by the user', async () => {
-        const user = userEvent.setup();
-        const AppComponent = render(<App />);
-        const AppDOM = AppComponent.container.firstChild;
-
-        const CitySearchDOM = AppDOM.querySelector('#city-search');
-        const CitySearchInput = within(CitySearchDOM).queryByRole('textbox');
-
-        await user.type(CitySearchInput, 'Berlin');
-        const berlinSuggestionItem = within(CitySearchDOM).queryByText('Berlin, Germany');
-        await user.click(berlinSuggestionItem);
-
-        const EventListDOM = AppDOM.querySelector('#event-list');
-        console.log(Object.keys(EventListDOM)) 
-/*         console.log(Object.keys(EventListDOM['__reactProps$vjrzr22ft8'])) 
- */
-        let allRenderedEventItems = within(EventListDOM).queryAllByRole('listitem');
-        allRenderedEventItems = allRenderedEventItems.filter(
-            event => event.textContent.includes('Berlin, Germany')
-        );
-        const allEvents = await getEvents();
-       
-        const berlinEvents = allEvents.filter(
-            event => event.location === 'Berlin, Germany'
-        );
-        
-         expect(allRenderedEventItems.length).toBe(berlinEvents.length);
- 
-        allRenderedEventItems.forEach(event => {
-            expect(event.textContent).toContain('Berlin, Germany');
-        });
+describe("<App /> integration", () => {
+  test("renders a list of events matching the city selected by the user", async () => {
+    const user = userEvent.setup();
+    const AppComponent = render(<App />);
+    const AppDOM = AppComponent.container.firstChild;
+
+    const CitySearchDOM = AppDOM.querySelector("#city-search");
+    const CitySearchInput = within(CitySearchDOM).queryByRole("textbox");
+
+    await user.type(CitySearchInput, "Berlin");
+    const berlinSuggestionItem =
+      within(CitySearchDOM).queryByText("Berlin, Germany");
+    await user.click(berlinSuggestionItem);
+
+    const EventListDOM = AppDOM.querySelector("#event-list");
+
+    let allRenderedEventItems = within(EventListDOM).queryAllByRole("listitem");
+    const filteredRenderedEventItems = allRenderedEventItems.filter((event) => {
+      return event.textContent.includes("Berlin, Germany");
+    });
+    const allEvents = mockData;
+
+    const berlinEvents = allEvents.filter((event) => {
+      return event.location === "Berlin, Germany";
     });
+
+    expect(allRenderedEventItems.length).toBe(berlinEvents.length);
```

Comment:
This test was failing because you App.js has a restriction to only render 32 items but `const allEvents = mockData;` returns 35 element and your filter on line 49 includes the 3 elements removed in the App.js so I commented them in the mock-data.js file